features/telemetry/distributed/common/src/main/java/org/opennms/netmgt/telemetry/distributed/common/PropertyTree.java

```diff
@@ -159,6 +160,39 @@ public String getRequiredString(final String... path) {
                 .orElseGet(Collections::emptyMap);
     }
 
+    /**
+     * Get the values of all child nodes at the given path, enumerating any subtrees and converting their
+     * paths into dot-separated property names. This is intended to handle parameter keys containing dots -
+     * see NMS-12738.
+     *
+     * @param path the path of the node to return split into its elements
+     * @return the {@link Map} from the children node names to its values. If one of the nodes in the path does not exist this will return an empty {@link Map}
+     */
+
+    public Map<String, String> getMapR(final String... path) {
+        Map<String,String> outmap = new HashMap<String,String>();
+        Optional<Node> node = this.find(path);
+        if(node.isPresent()) {
+            for(String k: node.get().children.keySet()) {
+                getChildKeys(node.get().children.get(k), k, outmap);
+            }
+        }
+        return outmap;
+    }
+
+    public void getChildKeys(Node n, String s, Map<String,String> m) {
+        if(n.value.isPresent()) {
+            m.put(s, n.value.get());
+            return;
```

Comment:
I'm not sure if this early return is correct. Consider the following keys:
* `a.b = 5`
* `a.b.c = 6`

This would skip the later value.

Reply:
Good catch! I've refactored in a more functional style and removed the early return. New test cases cover a parent key with a value.